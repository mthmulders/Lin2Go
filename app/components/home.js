import React, { Component, PropTypes } from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';
import { connect } from 'react-redux';
import Button from 'apsl-react-native-button';

import { startGame } from '../actions';
const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: 'red',
    justifyContent: 'center',
    fontSize: 18,
    marginLeft: 20,
    marginRight: 20
  },
  logo: {
    marginBottom: 40
  }
});

const Home = (props) => (
  <View>
    <Image source={ require('../resources/start.png') } style={ styles.logo } />
    <Button style={ styles.button } onPress={ props.startGame }>
      <Text>Start new game</Text>
    </Button>
  </View>
);

const mapStateToProps = (state) => {
  return {
    gameRunning: state.game !== null
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    startGame: () => dispatch(startGame())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
