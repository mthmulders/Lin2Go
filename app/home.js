import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';

import styles from './styles'

export default function (props) {
  return (
    <View style={ styles.container }>
      <Text>LIN2GO</Text>
    </View>
  );
};
