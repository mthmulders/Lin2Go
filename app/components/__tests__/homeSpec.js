jest.unmock('../home');

import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import { Home } from '../home';

describe('The home screen', () => {
  it('should show menu buttons', () => {
    // Arrange

    // Act
    const tree = renderer.create(<Home />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
