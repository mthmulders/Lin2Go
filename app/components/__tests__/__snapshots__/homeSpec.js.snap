exports[`The home screen should show menu buttons 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "red",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontSize": 18,
        "height": 44,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "paddingLeft": 20,
        "paddingRight": 20
      }
    }
    testID={undefined}>
    <Text>
      Start new game
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "red",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontSize": 18,
        "height": 44,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1,
        "paddingLeft": 20,
        "paddingRight": 20
      }
    }
    testID={undefined}>
    <Text>
      Statistics
    </Text>
  </View>
</View>
`;
